src/parser/mod.rs

```diff
@@ -397,11 +397,44 @@ fn pop_operator<'c>(operator_stack: &mut Vec<&Token>, results: &mut Vec<(Ast<'c>
     let (rhs, rhs_location) = results.pop().unwrap();
     let (lhs, lhs_location) = results.pop().unwrap();
     let location = lhs_location.union(rhs_location);
-    let operator = Ast::operator(operator_stack.pop().unwrap().clone(), location);
-    let call = Ast::function_call(operator, vec![lhs, rhs], location);
+    let operator = operator_stack.pop().unwrap().clone();
+    let call =  desugar_apply_operator(operator, lhs, rhs, location);
     results.push((call, location));
 }
 
+fn desugar_apply_operator<'a>(operator: Token, lhs: Ast<'a>, rhs: Ast<'a>, location: Location<'a>) -> Ast<'a> {
+    match operator {
+        Token::ApplyLeft => 
+            if let Ast::FunctionCall(fun) = lhs {
+                let mut args = vec![rhs];
```

Comment:
Arg vec handling mentioned

Reply:
I think something like the below would work (untested):
```rust
fn desugar_apply_operator<'a>(operator: Token, lhs: Ast<'a>, rhs: Ast<'a>, location: Location<'a>) -> Ast<'a> {
    match operator {
        Token::ApplyLeft => add_argument_to_function(lhs, rhs, location),
        Token::ApplyRight => add_argument_to_function(rhs, lhs, location),
        _ => {
            let operator = Ast::operator(operator, location);
            Ast::function_call(operator, vec![lhs, rhs], location)
        }
    }
}

fn add_argument_to_function<'a>(f: Ast<'a>, arg: Ast<'a>, location: Location<'a>) -> Ast<'a> {
    if let Ast::FunctionCall(mut call) = f {
        call.args.insert(0, arg);
        Ast::FunctionCall(call)
    } else {
        Ast::function_call(f, vec![arg], location)
    }
}
```

Reply:
Maybe that function could also be called `prepend_argument_to_function` for extra clarity.

Reply:
Thank you, missed the `mut call` so I thought I had to allocate a new vec